mozci/console/commands/push.py

```diff
@@ -211,6 +223,38 @@ def handle(self) -> None:
                     f"Classification and regressions details for push {push.push_uuid} were saved in {filename} JSON file"
                 )
 
+            # Send a notification when some emails are declared in the config
+            emails = config.get("emails")
+            if emails:
+
+                # Load previous classification from taskcluster
+                try:
+                    previous = push.get_existing_classification()
+                except SourcesNotFound:
+                    # Nothing to do if there is no pre-existing classification
+                    return
+
+                # Send an email notification when:
+                # - the previous classification was GOOD or UNKNOWN and the new classification is BAD;
+                # - or the previous classification was BAD and the new classification is GOOD or UNKNOWN.
+                if (
+                    previous in (PushStatus.GOOD, PushStatus.UNKNOWN)
+                    and classification == PushStatus.BAD
+                ) or (
+                    previous == PushStatus.BAD
+                    and classification in (PushStatus.GOOD, PushStatus.UNKNOWN)
+                ):
+                    send_admin_emails(
+                        emails=emails,
+                        subject=f"Push status evolution {push.id} {push.rev[:8]}",
+                        content=EMAIL_PUSH_EVOLUTION.format(
+                            previous=previous,
+                            classification=classification,
+                            push=push,
+                            real_failures="\n - ".join(regressions.real.keys()),
```

Comment:
What exactly is this printing? Just the task names or also the group names?

Reply:
Just the group name

Reply:
Ok, let's print both the task names and the group names that are failing, so it's easier for the receiver to see it.

We could potentially even give a treeherder link to the exact failure, like this:
https://github.com/mozilla/code-review/blob/9dc5e90e4f011e5b157ef679457b91e38d9bdc37/events/code_review_events/bugbug_utils.py#L394
https://github.com/mozilla/code-review/blob/9dc5e90e4f011e5b157ef679457b91e38d9bdc37/tools/code_review_tools/treeherder.py#L8
(treeherder link can be a follow-up if you want)

Reply:
For example, for the failure browser/extensions/formautofill/test/browser/creditcard/browser.ini on the push a867a89b52a6255e93a568ecc70cf6e0fe2387e0, we can do something like this:

# Real failures
- Group [group1](https://treeherder.mozilla.org/jobs?repo=autoland&tochange=a867a89b52a6255e93a568ecc70cf6e0fe2387e0&test_paths=browser%2Fextensions%2Fformautofill%2Ftest%2Fbrowser%2FcreditCard%2Fbrowser.ini) - Tasks [test-linux1804-64-qr/opt-mochitest-browser-chrome-swr-e10s-7](https://treeherder.mozilla.org/jobs?repo=autoland&revision=a867a89b52a6255e93a568ecc70cf6e0fe2387e0&selectedTaskRun=ehq3v2xQT0CfDvmcJ5UsbQ.0), [test-linux1804-64-qr/opt-mochitest-browser-chrome-spi-nw-e10s-7
](https://treeherder.mozilla.org/jobs?repo=autoland&revision=a867a89b52a6255e93a568ecc70cf6e0fe2387e0&selectedTaskRun=bVo5cDpGR3mPsMslKmZQLQ.0)

In the first link, tochange is a867a89b52a6255e93a568ecc70cf6e0fe2387e0, but actually we should use the push revision where the group actually run (in case it is not the current push like in this case but a following-one). We can start simple and just put the current push revision though.